tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/DefaultModelBindingContext_SetResult_Integration.cs

```diff
@@ -58,43 +60,46 @@ internal static CallTargetReturn OnMethodEnd<TTarget>(TTarget instance, System.E
                 {
                     if (defaultModelBindingContext.Result.IsModelSet && defaultModelBindingContext.IsTopLevelObject)
                     {
-                        var span = (state.Scope ?? state.PreviousScope).Span;
+                        var span = (state.Scope ?? state.PreviousScope)?.Span;
 
-                        if (defaultModelBindingContext.BindingSource.Id == "Body")
+                        if (span is not null)
```

Comment:
As I understand it (it's hard to follow in GitHub), if `span` is null, we can't do anything in this method? In which case, I would be inclined to add this as a short-circuit, right at the top of the method, as it avoids somewhat expensive procedures and should make it easier to read...

e.g. right at the top, doing a 
```csharp
var span = (state.Scope ?? state.PreviousScope)?.Span;
if (span is not null)
{
    return CallTargetReturn.GetDefault();    
}
```

WDYT?

Reply:
Thanks! Yes, since everything that is done beyond this point needs a span, we can safely skip that if we have no span. I have updated the code with the early return to make the code easier to read.